File: classify-server/src/main/java/io/ermdev/alice/entity/Student.java
package io.ermdev.alice.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import javax.xml.bind.annotation.XmlTransient;
import java.util.ArrayList;
import java.util.List;

@Table(name="tblstudent")
@Entity
public class Student {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String firstName;

    private String lastName;

    @OneToMany(mappedBy = "student", cascade = CascadeType.ALL)
    private List<Class> classes = new ArrayList<>();

    public Student(){}

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    @JsonIgnore
    @XmlTransient
    public List<Class> getClasses() {
        return classes;
    }

    public void setClasses(List<Class> classes) {
        this.classes = classes;
    }
}

File: classify-rest-webservice/src/main/java/io/classify/data/entity/Student.kt
package io.classify.data.entity

import javax.persistence.*

@Entity
@Table(name = "tbl_student")
class Student(@Id
              @GeneratedValue(strategy = GenerationType.AUTO)
              var id: Long = 0,

              var number: Long = 0,

              var name: String = "",

              @OneToOne(cascade = [(CascadeType.ALL)])
              @JoinColumn(name = "user_id", unique = true)
              var user: User = User(),

              @ManyToMany(mappedBy = "students")
              var lessons: List<Lesson> = ArrayList())